libs/wingc/src/jsify.rs

```diff
@@ -693,7 +695,90 @@ impl<'a> JSifier<'a> {
 						"\")"
 					)
 				}
-				_ => new_code!(expr_span, ""),
+				IntrinsicKind::Inflight => {
+					let arg_list = intrinsic.arg_list.as_ref().unwrap();
+					let path = if let ExprKind::Literal(Literal::String(ss)) = &arg_list.pos_args[0].kind {
+						// trim off the first and last character (the quotes)
+						let ss = &ss[1..ss.len() - 1];
+						let extern_path = Utf8Path::new(&ss);
+
+						let extern_path = normalize_path(extern_path, ctx.source_path);
+						if !extern_path.exists() {
+							todo!()
+							// self.spanned_error(exp, format!("Extern file '{}' does not exist", extern_path));
+						}
+						extern_path
+					} else {
+						return new_code!(&expression.span, "");
+					};
+
+					let mut export_name = "default";
+
+					let mut lifts: IndexMap<String, (&Expr, Option<&Vec<Expr>>)> = IndexMap::new();
+					for x in &arg_list.named_args {
+						if x.0.name == "export" {
+							if let ExprKind::Literal(Literal::String(s)) = &x.1.kind {
+								export_name = &s[1..s.len() - 1];
+							}
+						} else if x.0.name == "lifts" {
+							if let ExprKind::JsonLiteral { element, .. } = &x.1.kind {
+								if let ExprKind::JsonMapLiteral { fields } = &element.kind {
+									for field in fields {
+										let alias = field.0.name.clone();
+										if let ExprKind::JsonLiteral { element, .. } = &field.1.kind {
+											if let ExprKind::JsonMapLiteral { fields } = &element.kind {
```

Comment:
I can probably get this specific case to work when the object is pointing to a Json literal, but it's worth noting that lifts still need to be statically known.

Reply:
After some attempts it looks like even getting this case to work would take some major changes so I'll leave it to a future improvement.
Honestly I'd rather skip to figuring out a way to allow dynamic lifts.

Reply:
What's dynamic lifts?

Reply:
not requiring statically-known arguments

Reply:
Can you elaborate? What do you mean by statically known arguments?

Reply:
Kinda contrived example:

```wing
class SomeStuff {
  new(path: str, lifted: std.LiftDeclaration) {
    @inflight(path, lift: [lifted])
  }
}
```

When I say static I mean "while generating js" or "pre-preflight". During this time is when we have all the type information available to generate .d.ts files.

In the above example, the compiler doesn't know value `path` has so it can't immediately write the type info to disk.
In addition, inside `lifted` is the type of the object being lifted, the alias it will have, and the operations needed. All of these things are needed to generate the types but they will not be available to the compiler.

imo the solution is we need a preflight API for type information and then all this generation can happen during preflight.